File: ui/hub-mk3-ui/src/store/orm/segment-choice-arrangement.js
// Copyright (c) XJ Music Inc. (https://xj.io) All Rights Reserved.
import {attr, fk, Model} from "redux-orm";

class SegmentChoiceArrangement extends Model {
  toString() {
    return `SegmentChoiceArrangement: ${this.name}`;
  }

  // Declare any static or instance methods you need.
}

SegmentChoiceArrangement.modelName = 'SegmentChoiceArrangement';

// Declare your related fields.
SegmentChoiceArrangement.fields = {
  id: attr(),
  segmentId: fk({
    to: 'Segment',
    as: 'segment',
    relatedName: 'choiceArrangements',
  }),
  segmentChoiceId: fk({
    to: 'SegmentChoice',
    as: 'segmentChoice',
    relatedName: 'arrangements',
  }),
  programVoiceId: fk({
    to: 'ProgramVoice',
    as: 'programVoice',
    relatedName: 'segmentChoiceArrangements',
  }),
  instrumentId: fk({
    to: 'Instrument',
    as: 'instrument',
    relatedName: 'segmentChoiceArrangements',
  }),
};

export default SegmentChoiceArrangement;

File: engine/src/entities/segment/SegmentChoiceArrangement.cpp
// Copyright (c) XJ Music Inc. (https://xjmusic.com) All Rights Reserved.

#include "xjmusic/entities/segment/SegmentChoiceArrangement.h"

namespace XJ {

  bool SegmentChoiceArrangement::equals(const SegmentChoiceArrangement &segmentChoiceArrangement) const {
    return id == segmentChoiceArrangement.id &&
           segmentId == segmentChoiceArrangement.segmentId &&
           segmentChoiceId == segmentChoiceArrangement.segmentChoiceId &&
           programSequencePatternId == segmentChoiceArrangement.programSequencePatternId;
  }

  unsigned long long SegmentChoiceArrangement::hashCode() const {
    return std::hash<std::string>{}(id) ^
           std::hash<int>{}(segmentId) ^
           std::hash<std::string>{}(segmentChoiceId) ^
           std::hash<std::string>{}(programSequencePatternId);
  }

}// namespace XJ
